apps/web/src/server/applications/create/key-dates/__tests__/applications-create-key-dates.test.js

```diff
@@ -0,0 +1,55 @@
+import { parseHtml } from '@pins/platform';
+import supertest from 'supertest';
+import { createTestApplication } from '../../../../../../testing/index.js';
+
+const { app, installMockApi, teardown } = createTestApplication();
+const request = supertest(app);
+
+describe('Applications create key dates', () => {
+	beforeEach(installMockApi);
+	afterEach(teardown);
+
+	beforeEach(async () => {
+		await request.get('/applications-service/case-officer');
+	});
+
+	const baseUrl = '/applications-service/create-new-case/123/key-dates';
+
+	describe('GET /key-dates', () => {
+		it('should render the page', async () => {
+			const response = await request.get(baseUrl);
+			const element = parseHtml(response.text);
+
+			expect(element.innerHTML).toMatchSnapshot();
+			expect(element.innerHTML).toContain('Save and continue');
+		});
+	});
+
+	describe('POST /key-dates', () => {
+		describe('Web-side validation:', () => {
+			it('should show validation errors if dates are not valid', async () => {
+				const response = await request.post(baseUrl).send({
+					firstNotifiedDay: 'not_a_number',
+					submissionDay: 32
+				});
+				const element = parseHtml(response.text);
+
+				expect(element.innerHTML).toMatchSnapshot();
+				expect(element.innerHTML).toContain('Enter a valid day for the first notified date');
+				expect(element.innerHTML).toContain('Enter a valid day for the submission date');
```

Comment:
Is the error messaging "enter a valid day", when we are expecting a string that might just contain the quarter or something?

Reply:
ok. apparently there a two designs versions for this page, the correct one that you're talking about and the other.
guess what I was looking at?